RDKEMW-7209, RDKOSS-477: Update AppArmor userspace to 3.1.7

Lasya-Prakarsha-D-V opened the pull request:
Reason for change: Updates to a newer branch of the AppArmor userspace (mainly compiler), expanding features and opening opportunities to use new mediation types down the line
Test Procedure: check apparmor_parser, ensure version 3.1.7, verify all profiles are loaded correctly with no syntax errors, verify no new ALLOWED/DENIED/AUDIT logs
Risks: Possibly new mediation types are enforced (network mainly),
otherwise Low
Priority: P1

Commit:
RDKEMW-7209: Update AppArmor userspace to 3.1.7

Reason for change: Updates to a newer branch of the AppArmor
userspace (mainly compiler), expanding features and opening
opportunities to use new mediation types down the line

Test Procedure: check apparmor_parser, ensure version 3.1.7,
verify all profiles are loaded correctly with no syntax errors,
verify no new ALLOWED/DENIED/AUDIT logs
Risks: Possibly new mediation types are enforced (network mainly),
otherwise Low

Signed-off-by: ldonth501 <LasyaPrakarsha_DonthiVenkata@comcast.com>

Commit:
RDKEMW-7209, RDKOSS-477: Converting the recipe to work with newer bitbake

Reason for change:
Facing error - contains an operation using the old override syntax.
Please convert this layer/metadata before attempting to use with a newer bitbake.
Test Procedure: apparmor should be build and consumed successfully without errors.
Risks: Low
Priority: P1

Change-Id: I564629e4ec3777c525360fee0c907041b1d5d31e
Signed-off-by: ldonth501 <LasyaPrakarsha_DonthiVenkata@comcast.com>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/rdkcentral/meta-rdk-oss-reference?pullRequest=207) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/rdkcentral/meta-rdk-oss-reference?pullRequest=207) before we can accept your contribution.<br/><hr/>**ldonth501** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/rdkcentral/meta-rdk-oss-reference?pullRequest=207) it.</sub>

rdkcmf-jenkins commented:
b'## **Blackduck scan failure details**

**Summary**: 0 **violation**s, 0 files pending approval, 1 file pending identification.
* Protex Server Path: /home/blackduck/github/meta-rdk-oss-reference/207/rdkcentral/meta-rdk-oss-reference

* Commit: 0777a8ac7d13f08484d5815c40a8041415cc8b60

 **Report detail:** [gist](https://gist.github.com/rdkcmf-jenkins/43d25dff82df530a014b6c105ff603b1)'

Lasya-Prakarsha-D-V committed:
Merge pull request #208 from rdkcentral/develop

Rebase with develop

Lasya-Prakarsha-D-V committed:
RDKEMW-7209, RDKOSS-477: Update AppArmor userspace to 3.1.7

Reason for change: Setting PREFERRED_VERSION of apparmor to 3.1.7 
Test Procedure: apparmor should be build and consumed successfully without errors.
Risks: Low
Priority: P1

Commit:
RDKEMW-7209, RDKOSS-477: Removing apparmor old recipe 2.13.10 as per OSS guidelines

Reason for change: As per OSS guidelines, every recipe/component maintaning single version,
Hence removing older version of recipe
Test Procedure: apparmor should be build and consumed successfully without errors.
Risks: Low
Priority: P1

Signed-off-by: ldonth501 <LasyaPrakarsha_DonthiVenkata@comcast.com>

Commit:
RDKEMW-7209, RDKOSS-477: Removing Preferred version 3.1.7 of apparmor

Reason for change: Since removed old recipe 2.13.10 version,
need not maintain PREFERRED_VERSION of apparmor to 3.1.7.
Test Procedure: apparmor should be build and consumed successfully without errors.
Risks: Low
Priority: P1

Signed-off-by: ldonth501 <LasyaPrakarsha_DonthiVenkata@comcast.com>

Ulrond commented:
@rdke_ghec_meta_oss_maintainer -> This should be rejected, there's no "Why", which is the first requirement that needs to be assessed. It's not enough that someone would like it... That's not a "Why" you need a real reason for risking the platform and causing the upgrade, As part of the governance.

maniselva006c committed:
Merge branch 'develop' into topic/RDKEMW-7209
